third-party/crowdsec/sample-configuration/parsers/james-auth.yaml

```diff
@@ -0,0 +1,27 @@
+onsuccess: next_stage
+debug: true
+filter: "evt.Parsed.program == 'james'"
+name: linagora/james-connect-security-log
+description: "Parser for James Connect Security Log"
+
+pattern_syntax:
+  JAMES_AUTH_FAIL: '\{"timestamp":"%{DATA:time}","level":"%{LOGLEVEL:level}","thread":"%{DATA:thread}","mdc":{"protocol":"%{DATA:protocol}","ip":"%{IPV4:ip}","host":"%{IPV4:host}","action":"LOGIN","login-user":"%{EMAILADDRESS:loginUser}","sessionId":"%{DATA:sessionId}","user":"%{DATA:user}"\},"logger":"%{JAVACLASS:logger}","message":"User %{DATA:loginUser} failed authentication","context":"%{DATA:context}"\}'
```

Comment:
Not sure it is a good idea to rewrite a JSON parser with a REGEXP... 
How robust is this solution to field reordering for instance?

Reply:
Because Crowdsec doesn't support to read JSON format so I have to read it with a Regex. 
IMO, it is seldom when the field reorder but maybe there will be some new added fields in the future so I will reconsider this pattern syntax

Reply:
Did you open a bug report for your issue with JSON format?

> IMO, it is seldom when the field reorder but maybe there will be some new added fields in the future so I will reconsider this pattern syntax

We already know it will break hard in the future. We should reconsider it now.

What does it take to write a parser in CrowdSec?

Reply:
From the mob / brainstorming, actually maybe two leads:

- from tung: we could add via logback an other entryof logs just for crowdsec with just the fields we are interested in in the format we want. Then we isolate the ones with AUTH_FAIL in crowdsec pattern stuff
- from rene:  we could maybe have a pattern syntax like `'%{DATA:data}failed authentication%{DATA:data}'` to just get logs with failed auth and then try to see if can use https://docs.crowdsec.net/docs/expr/helpers/#json-helpers to extract in a json way the fields we want from the logs